File: src/Java/Group.java
package Java;
import java.util.ArrayList;
import java.util.List;

public class Group {
    private List<User> members;
    private List<User> admins;
    private List<String> messages;
    private List<Post> posts;
    private String name;
    private String description;
    private List<Group> subgroups;
    private final int MAX_SUBGROUPS =5;
    private final int MAX_ADMINS = 5;
    private List<User> requestsUsers; 

    public Group(){}

    public Group(String name, String description){
        this.name = name;
        this.description = description;
        this.members = new ArrayList<>();
        this.admins = new ArrayList<>(MAX_ADMINS);
        this.posts = new ArrayList<>();
        this.messages = new ArrayList<>();
        this.subgroups = new ArrayList<>(MAX_SUBGROUPS);
        this.requestsUsers = new ArrayList<>();
    }

    // Metodos Getters
    public List<User> getMembers(){
        return this.members;
    }

    public List<User> getAdmins(){
        return this.admins;
    }

    public List<Post> getPosts(){
        return this.posts;
    }

    public String getName(){
        return this.name;
    }

    public String getDescription(){
        return this.description;
    }

    public List<Group> getSubgroups(){
        return this.subgroups;
    }

    public List<User> getRequestsUsers(){
        return this.requestsUsers;
    }

    // Metodos Setters
    public void setMembers(List<User> members){
        this.members = members;
    }

    public void setAdmins(List<User> admins){
        this.admins = admins;
    }

    public void setName(String name){
        this.name = name;
    }   

    public void setDescription(String description){
        this.description = description;     
    }

    public void addAdmin(User user){
        this.admins.add(user);
    }

    public void addMember(User user){
        this.members.add(user);
    }

    public void addPost(Post post){
        this.posts.add(post);
    }

    public void addSubgroup(Group group){
        this.subgroups.add(group);
    }

    public void addMessage(String message){
        this.messages.add(message);
    }

    public void listMembers(){
        System.out.println("-------------------------------------");
        System.out.println("Listado de Miembros:\n");
        for(User u: this.members){
            System.out.println(u);
        }
    }

    @Override
    public String toString(){
        StringBuilder sb = new StringBuilder();
        sb.append("Grupo: \n");
        sb.append("-----------------------------------\n");
        sb.append("Nombre: " + this.name + "\n");
        sb.append("Descripción: " + this.description + "\n");
        sb.append("Membres: \n");
        for(User u: this.members){
            sb.append(u);
        }
        sb.append("-------------------------------------");
        return sb.toString();
    }

    public void addRequest(User user){
        this.requestsUsers.add(user);
    }

    public void listRequests(){
        System.out.println("-------------------------------------");
        System.out.println("Listado de Solicitudes:\n");
        for(User u: this.requestsUsers){
            System.out.println(u);
        }
    }



}

File: src/C#/networking-app/MyApplication/Group.cs

using System.Text;

namespace MyApplication.Models
{
	public class Group
	{
		private string name;
		private string description;
		private List<User> members;
		private List<User> admins;
		private List<Post> posts;
		private List<Group> subGroups;
		private List<User> requestsUsers;
		private List<string> messages;
		private const int MAX_SUBGROUPS = 5;
		private const int MAX_ADMINS = 5;


		public Group(string name, string description)
		{
			this.name = name;
			this.description = description;
			this.members = new List<User>();
			this.admins = new List<User>(MAX_ADMINS);
			this.posts = new List<Post>();
			this.subGroups = new List<Group>(MAX_SUBGROUPS);
			this.requestsUsers = new List<User>();
			this.messages = new List<string>();
		}

		// Metodos Getters y Setters
		public string Name
		{
			get { return name; }
			set { name = value; }
		}

		public string Description
		{
			get { return description; }
			set { description = value; }
		}

		public List<User> Members
		{
			get { return members; }
		}

		public List<User> Admins
		{
			get { return admins; }
		}

		public List<Post> Posts
		{
			get { return posts; }
		}

		public List<Group> SubGroups
		{
			get { return subGroups; }
		}

		public List<User> RequestsUsers
		{
			get { return requestsUsers; }
		}

		public List<string> Messages
		{
			get { return messages; }
		}

		public void AddAdmin(User user)
		{
			this.admins.Add(user);
		}

		public void AddMember(User user)
		{
			this.members.Add(user);
		}

		public void AddPost(Post post)
		{
			this.posts.Add(post);
		}

		public void AddSubgroup(Group group)
		{
			this.subGroups.Add(group);
		}

		public void AddMessage(string message)
		{
			this.messages.Add(message);
		}

		public void ListMembers()
		{
			Console.WriteLine("-------------------------------------");
			Console.WriteLine("Listado de Miembros:\n");
			foreach (User u in this.members)
			{
				Console.WriteLine(u);
			}
		}

		public override string ToString()
		{
			StringBuilder sb = new StringBuilder();
			sb.Append("Grupo: \n");
			sb.Append("-----------------------------------\n");
			sb.Append("Nombre: " + this.name + "\n");
			sb.Append("Descripción: " + this.description + "\n");
			sb.Append("Membres: \n");
			foreach (User u in this.members)
			{
				sb.Append(u);
			}
			sb.Append("-------------------------------------");
			return sb.ToString();

		}

		public void AddRequest(User user)
		{
			this.requestsUsers.Add(user);
		}

		public void ListRequests()
		{
			Console.WriteLine("-------------------------------------");
			Console.WriteLine("Listado de Solicitudes:\n");
			foreach (User u in this.requestsUsers)
			{
				Console.WriteLine(u);
			}
		}

	}
}
